sql/2020/19_Compression/19_02.byte_saving_of_text_compression.sql

```diff
@@ -0,0 +1,20 @@
+#standardSQL
+#Text Based Compression Byte Savings
+SELECT 
+  _TABLE_SUFFIX AS client,
+  IF(JSON_EXTRACT_SCALAR(report, "$.audits.uses-text-compression.score") != "1", 'compression', 'non_compression') AS compression,
+  percents,
+  APPROX_QUANTILES(( CAST(JSON_EXTRACT_SCALAR(report, "$.audits.uses-text-compression.details.overallSavingsBytes") AS INT64) / 1024), 1000)[OFFSET(percents * 10)] AS kbyte_savings,
+  COUNT(0) AS num_pages,
+  SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS total,      
```

Comment:
The cross join of the lighthouse table with the unnested percents means that there are ~30 million rows in the results, so `COUNT(0)` won't be counting pages it'll be counting 5x per page.

Since 19_04 is also counting the percent of pages by this audit's score, I think it can be removed from here.

Reply:
nice explanation. thanks.